tls/s2n_recv.c

```diff
@@ -213,6 +213,17 @@ ssize_t s2n_recv(struct s2n_connection * conn, void *buf, ssize_t size, s2n_bloc
 }
 
 uint32_t s2n_peek(struct s2n_connection *conn) {
+    if (conn == NULL) {
```

Comment:
This check will silently fail when the user provides wrong input. It is better to return error (like -1 if it makes sense) or just to fail with exception as of before. returning 0 seems wrong and not aligned with other functions in this file.

Reply:
Unfortunately we can't return -1 here since that would require changing the return type, which could potentially break existing callers. The way it is now it will segfault which is unrecoverable. A compromise might be that we use [assert](https://en.cppreference.com/w/c/error/assert) before returning 0, which is easier to understand than a segfault. It also has the benefit that it's removed when NDEBUG is defined.

Reply:
I'm a little leery of using assert here, because it's not used anywhere else in the codebase.

Reply:
Fully understand the concern with changing return types. Still, I would prefer segfault on top of zero here. Segfault will clearly raise the issue to the app developer while zero will be consumed by the app business logic and will take the app to the wrong conclusion. Now, I do understand that this is library wide design question, so it is better to be decided as such and not in specific discussion about this API. Looking at other APIs I thought the (implicit or explicit) decision was to go with segfault so I'm worried about changing this here on this API.